src/configure/helper/repoAnalysisHelper.ts

```diff
@@ -1,34 +1,45 @@
 import * as path from 'path';
-import { PortalExtensionClient } from "../clients/portalExtensionClient";
+import { ModaRepositoryAnalysisClient } from '../clients/modaRepositoryAnalysisClient';
+import { PortalExtensionRepositoryAnalysisClient } from "../clients/portalExtensionRepositoryAnalysisClient";
 import { AzureSession, GitRepositoryParameters, RepositoryAnalysisApplicationSettings, RepositoryAnalysisParameters, RepositoryAnalysisRequest, RepositoryDetails, RepositoryProvider, SupportedLanguage } from "../model/models";
 import { RepoAnalysisConstants } from "../resources/constants";
+import { IServiceUrlDefinition, RemoteServiceUrlHelper, ServiceFramework } from './remoteServiceUrlHelper';
 
-export class RepoAnalysisHelper {
-    private portalExtensionClient: PortalExtensionClient;
 
-    constructor(azureSession: AzureSession) {
-        this.portalExtensionClient = new PortalExtensionClient(azureSession.credentials);
+export class RepoAnalysisHelper {
+    private azureSession: AzureSession;
+    private githubPatToken?: string;
+    constructor(azureSession: AzureSession, githubPatToken?: string) {
+        this.azureSession = azureSession;
+        this.githubPatToken = githubPatToken;
     }
 
     public async getRepositoryAnalysis(sourceRepositoryDetails: GitRepositoryParameters, workspacePath: string): Promise<RepositoryAnalysisParameters> {
 
-        //As of now this solution has support only for github
-        if (sourceRepositoryDetails.repositoryProvider != RepositoryProvider.Github) {
-            return null;
-        }
-
         let repositoryAnalysisResponse;
         try{
-            let repositoryDetails: RepositoryDetails = new RepositoryDetails();
+            const serviceDefinition = await RemoteServiceUrlHelper.getRepositoryAnalysisDefinition();
+            const client = this.getClient(serviceDefinition);
+
+            const repositoryDetails: RepositoryDetails = new RepositoryDetails();
             repositoryDetails.id = sourceRepositoryDetails.repositoryId;
             repositoryDetails.defaultbranch = !!sourceRepositoryDetails.branch ? sourceRepositoryDetails.branch : RepoAnalysisConstants.Master;
             repositoryDetails.type = RepositoryProvider.Github;
 
             let repositoryAnalysisRequestBody = new RepositoryAnalysisRequest;
             repositoryAnalysisRequestBody.Repository = repositoryDetails;
             repositoryAnalysisRequestBody.WorkingDirectory = workspacePath;
-
-            repositoryAnalysisResponse = await this.portalExtensionClient.getRepositoryAnalysis(repositoryAnalysisRequestBody);
+            repositoryAnalysisRequestBody.Repository.authorizationInfo = {
+                scheme: "Token",
+                parameters: {
+                    accesstoken: this.githubPatToken
+                }
+            };
+            if (serviceDefinition.serviceFramework === ServiceFramework.Vssf) {
```

Comment:
this if condition is not required anymore. that is the beauty of using an abstract interface :)

Reply:
Thanks!